src/Http/Http.Extensions/test/RequestDelegateFactoryTests.cs

```diff
@@ -4891,6 +4906,396 @@ void TestAction(IFormFile? file, TraceIdentifier traceId)
+        {
+            messageArgument = message;
+            traceIdArgument = traceId;
+        }
+
+        var stream = new MemoryStream();
+        await content.CopyToAsync(stream);
+
+        stream.Seek(0, SeekOrigin.Begin);
+
+        var httpContext = CreateHttpContext();
+        httpContext.Request.Body = stream;
+        httpContext.Request.Headers["Content-Type"] = contentType;
+        httpContext.Features.Set<IHttpRequestBodyDetectionFeature>(new RequestBodyDetectionFeature(true));
+        httpContext.TraceIdentifier = "my-trace-id";
+
+        var factoryResult = RequestDelegateFactory.Create(TestAction);
+        var requestDelegate = factoryResult.RequestDelegate;
+
+        await requestDelegate(httpContext);
+
+        Assert.Equal(httpContext.Request.Form["message"], messageArgument);
+        Assert.NotNull(messageArgument);
+
+        Assert.Equal("my-trace-id", traceIdArgument.Id);
+    }
+
+    [Fact]
+    public async Task RequestDelegatePopulatesFromBothIFormCollectionAndIFormFileParameters()
+    {
+        IFormFileCollection? formFilesArgument = null;
+        IFormCollection? formArgument = null;
+
+        void TestAction(IFormCollection form, IFormFileCollection formFiles)
+        {
+            formFilesArgument = formFiles;
+            formArgument = form;
+        }
+
+        var fileContent = new StringContent("hello", Encoding.UTF8, "application/octet-stream");
+        var form = new MultipartFormDataContent("some-boundary");
+        form.Add(fileContent, "file", "file.txt");
+        form.Add(new StringContent("foo"), "name");
+
+        var stream = new MemoryStream();
+        await form.CopyToAsync(stream);
+
+        stream.Seek(0, SeekOrigin.Begin);
+
+        var httpContext = CreateHttpContext();
+        httpContext.Request.Body = stream;
+        httpContext.Request.Headers["Content-Type"] = "multipart/form-data;boundary=some-boundary";
+        httpContext.Features.Set<IHttpRequestBodyDetectionFeature>(new RequestBodyDetectionFeature(true));
+
+        var factoryResult = RequestDelegateFactory.Create(TestAction);
+        var requestDelegate = factoryResult.RequestDelegate;
+
+        await requestDelegate(httpContext);
+
+        Assert.Equal(httpContext.Request.Form.Files, formFilesArgument);
+        Assert.NotNull(formFilesArgument!["file"]);
+
+        Assert.Equal(httpContext.Request.Form, formArgument);
+        Assert.NotNull(formArgument);
+        Assert.Collection(formArgument!,
+            (item) =>
+            {
+                Assert.Equal("name", item.Key);
+                Assert.Equal("foo", item.Value);
+            });
+
+        var allAcceptsMetadata = factoryResult.EndpointMetadata.OfType<IAcceptsMetadata>();
+        Assert.Collection(allAcceptsMetadata,
+            (m) => Assert.Equal(new[] { "multipart/form-data", "application/x-www-form-urlencoded" }, m.ContentTypes),
+            (m) => Assert.Equal(new[] { "multipart/form-data" }, m.ContentTypes));
```

Comment:
Huh, this test implies my earlier comment is wrong? How are both metadata added?

Reply:
I tried give you an explanation, I hope it is clearer.

Reply:
Oh I think I see what is happening, I believe if you change the test to be
```
void TestAction(IFormFileCollection formFiles, IFormCollection form)
```
(notice form and formFiles are now swapped), now there will be 1 accepts metadata.

Reply:
Exactly, because `FormFile` is a `From` but `From` is not a `FormFile `😂

Reply:
I'm lost D: